user/views.py

```diff
@@ -4,15 +4,23 @@
 from django.contrib import messages
 from .models import User
 from django.http import Http404
-
-
+from tweet.models import Tweet
+from django.core.paginator import Paginator,PageNotAnInteger,EmptyPage
 
 
 
 def home(request):
-  if request.user.is_authenticated:
-    return render(request,'home.html')
-  return redirect('user:newSession')
+  if not request.user.is_authenticated:
+    return redirect('user:newSession')
+  all_tweets = Tweet.objects.all().order_by('-id')
+  page = request.GET.get('page', 1)
+  paginator = Paginator(all_tweets, 1)
+  try:
+    tweets = paginator.page(page)
+  except (PageNotAnInteger, EmptyPage):
+    tweets = paginator.page(1)
+  return render(request,'home.html', {'all_tweets': tweets})
```

Comment:
- How are you going to page 2 ?
- Where is comfiguration how many tweets should be loaded per page

Reply:
- page 2 and other work from hompepage there is pagination
```python
 page = request.GET.get('page', 1)
``` this will get the page and if not found the default will be 1

- we can modify the below to load as many as we want, now it is set to 1
- 
```python
paginator = Paginator(all_tweets, 1)
```